src/read_canonical_selection.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include "read_bulk.hpp"
+
+namespace bbp {
+namespace sonata {
+namespace detail {
+
+template <class T, class XRange, class YRange>
+std::vector<T> readCanonicalSelection(const HighFive::DataSet& dset,
+                                      const std::vector<XRange>& xranges,
+                                      const std::vector<YRange>& yranges) {
+    if (yranges.size() != 1) {
+        throw SonataError("Only yranges.size() == 1 has been implemented.");
+    }
+
+    size_t j_begin = std::get<0>(yranges[0]);
+    size_t j_end = std::get<1>(yranges[0]);
+
+    constexpr size_t min_gap_size = SONATA_PAGESIZE / (2 * sizeof(uint64_t));
```

Comment:
will `SONATA_PAGESIZE` show up in other files? I'd prefer to move it to this one if it's only used here.

Reply:
If continue with the set of changes, then this function will move into a virtual method. Which means the constant can move into the class too, and be a runtime constant, not a compile time constant.

Reply:
Yup, sounds good.